estate/models/estate_property.py

```diff
@@ -141,3 +139,22 @@ def action_cancel(self):
             record.state = "canceled"
 
 
+    _sql_constraints = [
+        (
+            "check_expected_price",
+            "CHECK(expected_price > 0)",
+            "Expected price must be strictly positive",
+        ),
+        (
+            "check_selling_price",
+            "CHECK(selling_price > 0)",
+            "Expected price must be strictly positive",
+        )
+    ]
+
+    @api.ondelete(at_uninstall=False)
+    def prevent_delete(self):
```

Comment:
wrong method declaration

we should follow this:

>  Methods decorated with ``@ondelete`` should raise an error following some
    conditions, and by convention, the method should be named either
    ``_unlink_if_<condition>`` or ``_unlink_except_<not_condition>``.

```suggestion
    def unlink_if_not_new_or_cancelled(self):
```

maybe something like that